### PR TITLE
Added MinecraftInfo as well as wstring support

### PR DESCRIPTION
**Pull request by Adrian8115**

### CODE REVIEW

**Comment by Intoprelised** [comment]
Id suggest using winrt for the different string types conversion, functions such as `winrt::to_string`, `winrt::impl::to_wstring_view` and `winrt::to_hstring` (which can be converted to a wstring) can be quite useful here.

File: AmethystAPI/src/amethyst/Log.h [hunk 1]
```diff
@@ -14,18 +17,41 @@ namespace Log {
         fmt::print("{}\n", formatted_string);
     }
 
+    // For wstring
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/Log.h:20]: These for wstring comments aren't really necessary

File: AmethystAPI/src/amethyst/Utility.cpp [hunk 2]
```diff
@@ -1,4 +1,11 @@
 #include "amethyst/Utility.h"
+#include <codecvt>
+#include <locale>
+
+// std::wstring_convert<std::codecvt_utf8<wchar_t>> is deprecated
+// but there is no replacement for it currently (as of C++23)
+// TODO find a working replacement
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/Utility.cpp:8]: Look into what Intop suggested. It definitely is a thing you can do though, as I've done it in the past. This should be removed.

File: AmethystAPI/src/amethyst/Utility.h [hunk 3]
```diff
@@ -1,7 +1,8 @@
 #pragma once
 #include <string>
 
-/**
- * Returns the path to the amethyst UWP folder
-*/
-std::string GetAmethystFolder();
\ No newline at end of file
+/// Returns the path to the amethyst UWP folder
+std::string GetAmethystFolder();
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/Utility.h:5]: Why change the /** */ comment, im just curious? I've been using them throughout the project, so I think its good to stay consistent

File: AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#include "minecraft/src-deps/core/semVer/SemVersion.h"
+#include <string>
+#include <winrt/Windows.ApplicationModel.h>
+
+namespace Amethyst {
+    class MinecraftInfo {
+    public:
+        std::string name;
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h:8]: Member variables should be prefixed with m. E.g mName for consistency with the rest of the project

File: AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.cpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
+#include "MinecraftInfo.h"
+#include "amethyst/Log.h"
+#include "amethyst/Utility.h"
+
+Amethyst::MinecraftInfo::MinecraftInfo()
+{
+    // Get the current uwp package
+    auto w_package = winrt::Windows::ApplicationModel::Package::Current();
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.cpp:8]: Variable names should be using camelCase not a mix of snake_case with camelCase

**Comment by Adrian8115** [comment]
Thank you for your advice, I never knew winrt had such capabilities!

File: AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#include "minecraft/src-deps/core/semVer/SemVersion.h"
+#include <string>
+#include <winrt/Windows.ApplicationModel.h>
+
+namespace Amethyst {
+    class MinecraftInfo {
```
Comment (FrederoxDev) [inline AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h:6]: I'm not too fond on the MinecraftInfo name, as it is not very specific. Maybe something like MinecraftVersion would be better. Open to suggestions

**Review by FrederoxDev — CHANGES_REQUESTED** [review]

File: AmethystAPI/src/amethyst/Utility.h (hunk 3 above)
Comment (Adrian8115) [reply AmethystAPI/src/amethyst/Utility.h:5]: I just thought they were a bit weird, but I looked into the other files and I see the purpose

File: AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h (hunk 6 above)
Comment (Adrian8115) [reply AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h:6]: I talked to duck, we came up with `MinecraftPackageInfo`, might that be better?

Comment (FrederoxDev) [reply AmethystAPI/src/amethyst/runtime/minecraft/MinecraftInfo.h:6]: Yup that's great

**Comment by Adrian8115** [comment]
should be good now...

**Review by FrederoxDev — APPROVED** [review]
LGTM

**Pull request merged**